Remove duplicate nodeSelector from the udev discovery manifest

ammmze opened the pull request:
<!--  Thank you for contributing to the Akri repo! Before submitting this PR, please make sure:
1. Read the Contributing Guide before submitting your PR: https://docs.akri.sh/community/contributing
2. Decide whether you need to add any labels to your PR, such as `same version` if the version should not be changed and your change will trigger the version check workflow. This will cause the workflow to automatically succeed: https://github.com/project-akri/akri/blob/main/.github/workflows/check-versioning.yml
3. If this PR closes another issue, add 'closes #<issue number>' somewhere in the PR summary. GitHub will automatically close that issue when this PR gets merged. Alternatively, adding 'refs #<issue number>' will not close the issue, but help provide the reviewer more context. -->

**What this PR does / why we need it**:
The kube manifest for the udev-discovery-handler has 2 nodeSelector elements. It looks like they've both been there a while, but somehow recently (maybe newer version of helm in flux idk) I am getting this error when it tries to perform the helm upgrade.

```
Helm upgrade failed: error while running post render on files: yaml: unmarshal errors:
  line 51: mapping key "nodeSelector" already defined at line 31
```

The log didn't specify what file it was, but this was the only one I saw having duplicate nodeSelector fields, so I'm guessing it was this one. And the udev stuff is the only thing i'm running currently with akri.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR has an associated PR with documentation in [akri-docs](https://github.com/project-akri/akri-docs)
- [ ] this PR contains unit tests
- [ ] added code adheres to standard Rust formatting (`cargo fmt`)
- [ ] code builds properly (`cargo build`)
- [ ] code is free of common mistakes (`cargo clippy`)
- [ ] all Akri tests succeed (`cargo test`)
- [ ] inline documentation builds (`cargo doc`)
- [ ] all commits pass the [DCO bot check](https://probot.github.io/apps/dco/) by being signed off -- see the failing DCO check for instructions on how to retroactively sign commits

ammmze commented on deployment/helm/templates/udev-discovery-handler.yaml:
```diff
@@ -16,8 +16,6 @@ spec:
         app.kubernetes.io/name: akri-udev-discovery
         app.kubernetes.io/component: discovery-handler
     spec:
-      nodeSelector:
-        "kubernetes.io/os": linux
```
This is the change ... everything else was from the version bump

diconico07 reviewed (approved):
LGTM. 
The k8s e2e issue is likely the same I got (and fixed) in #670, so as soon as this one gets merged you should rebase to get the fix and get all tests green.

ammmze commented:
> LGTM. The k8s e2e issue is likely the same I got (and fixed) in #670, so as soon as this one gets merged you should rebase to get the fix and get all tests green.

Thank you! I've rebased and the checks all passed now!
